File: app/src/main/java/fr/geobert/radis/BaseFragment.java
package fr.geobert.radis;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import fr.geobert.radis.data.AccountManager;
import fr.geobert.radis.tools.UpdateDisplayInterface;

public abstract class BaseFragment extends Fragment implements UpdateDisplayInterface {
    protected MainActivity mActivity;
    protected AccountManager mAccountManager;
//    public boolean doOnResumeWhenReady = false;

    protected void baseInit() {
        this.mActivity = (MainActivity) getActivity();
        this.mAccountManager = mActivity.getAccountManager();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        baseInit();
        return null;
    }

//    @Override
//    public void onResume() {
//        super.onResume();
//        if (doOnResumeWhenReady) {
//            doOnResumeWhenReady = false;
//            doOnResume();
//        }
//    }
//
//    public abstract void processOnResume();
//
//    public final void doOnResume() {
//        if (isAdded()) {
//            processOnResume();
//        } else {
//            doOnResumeWhenReady = true;
//        }
//    }

    public abstract void onRestoreInstanceState(Bundle savedInstanceState);

    public abstract boolean onAccountChanged(long itemId);

    public String getName() {
        return this.getClass().getName();
    }
}

File: app/src/main/kotlin/fr/geobert/radis/BaseFragment.kt
package fr.geobert.radis

import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import fr.geobert.radis.data.AccountManager
import fr.geobert.radis.tools.UpdateDisplayInterface
import kotlin.properties.Delegates
import android.support.v7.widget.Toolbar

public abstract class BaseFragment : Fragment(), UpdateDisplayInterface, Toolbar.OnMenuItemClickListener {
    protected val mActivity: MainActivity by Delegates.lazy { getActivity() as MainActivity }
    protected val mAccountManager: AccountManager by Delegates.lazy { mActivity.mAccountManager }

    //    public abstract fun onRestoreInstanceState(savedInstanceState: Bundle)

    public abstract fun onOperationEditorResult(requestCode: Int, resultCode: Int, data: Intent?)

    public abstract fun onAccountChanged(itemId: Long): Boolean

    public fun getName(): String {
        return this.javaClass.getName()
    }

    protected fun setIcon(id: Int) {
        mActivity.mToolbar.setNavigationIcon(id)
    }

    protected fun setMenu(id: Int) {
        mActivity.mToolbar.getMenu().clear()
        mActivity.mToolbar.inflateMenu(id)
        mActivity.mToolbar.setOnMenuItemClickListener(this)
    }
}
